src/translate-virtual-address.h

```diff
@@ -148,26 +158,40 @@ static NO_INLINE bool translate_virtual_address(STATE_ACCESS &a, uint64_t *ppadd
     // Hence, the use of arithmetic shifts here
     int vaddr_bits = XLEN - (LOG2_PAGE_SIZE + levels * LOG2_VPN_SIZE);
     if (unlikely((static_cast<int64_t>(vaddr << vaddr_bits) >> vaddr_bits) != static_cast<int64_t>(vaddr))) {
-        return false;
+        *ppaddr = vaddr;
```

Comment:
From my understanding, you only need to set this on failure if `TRANSLATION_MODE == TRANSLATION_G`. (optimization)